### PR TITLE
scripts and tools: Make --setup command independent

### PR DESCRIPTION
**Pull request by hebasto**
This PR allows a user to run:
```sh
./gitian-build.py --setup
```
without unused `signer` and `version` options.

In master the `signer` and `version` options are mandatory. This implies the following code is dead:
https://github.com/bitcoin/bitcoin/blob/387eb5b34307448f16d3769ac0245c4e3d996a38/contrib/gitian-build.py#L192-L200
This PR fixes those lines of code.

Also this PR has a nice side effect: there is no more warnings about macOS build during processing `--setup` command. Ref: https://github.com/bitcoin/bitcoin/pull/13998#issuecomment-493691117

Note: https://github.com/bitcoin-core/docs/blob/master/gitian-building.md will be updated when this PR is merged.

### CODE REVIEW

File: contrib/gitian-build.py [hunk 1]
```diff
@@ -41,7 +41,6 @@ def setup():
     if args.is_bionic and not args.kvm and not args.docker:
         subprocess.check_call(['sudo', 'sed', '-i', 's/lxcbr0/br0/', '/etc/default/lxc-net'])
         print('Reboot is required')
-        exit(0)
```
Comment (laanwj) [inline contrib/gitian-build.py]: Why remove this line?

**Comment by laanwj** [comment]
I think this breaks the functionality of using `--setup` together with other options?

**Comment by hebasto** [comment]
> I think this breaks the functionality of using `--setup` together with other options?

Yes. This was made intended.

Currently, the script will stop regardless of other options ( `--build`, `--sign` etc):
https://github.com/bitcoin/bitcoin/blob/03b8596dd665d2f70c917794295911adb8680bcc/contrib/gitian-build.py#L41-L44

File: contrib/gitian-build.py (hunk 1 above)
Comment (hebasto) [reply contrib/gitian-build.py]: > Why remove this line?

See [my comment below](https://github.com/bitcoin/bitcoin/pull/15236#issuecomment-456756208).

File: contrib/gitian-build.py [hunk 2]
```diff
@@ -184,6 +177,16 @@ def main():
         if not 'LXC_GUEST_IP' in os.environ.keys():
             os.environ['LXC_GUEST_IP'] = '10.0.3.5'
 
+    if args.setup:
+        setup()
+        exit(0)
```
Comment (practicalswift) [inline contrib/gitian-build.py]: Should be `sys.exit(0)` :-)

Surprised that `flake8` didn't complain about that!

**Comment by laanwj** [comment]
> Currently, the script will stop regardless of other options ( --build, --sign etc):

It should because a reboot is intended in those specific cases, not always!

**Comment by hebasto** [comment]
@laanwj 

You are right. I mean "can stop", not "will stop".

It will stop using LXC on Bionic.

File: contrib/gitian-build.py (hunk 2 above)
Comment (practicalswift) [reply contrib/gitian-build.py]: Thanks!

**Comment by hebasto** [comment]
Also, a user do not want to run `--setup` together with other commands because after `--setup` they usually want to copy `MacOSX10.11.sdk.tar.gz` to `gitian-builder/inputs` directory before `--build`.

**Comment by ken2812221** [comment]
utACK 195054096aa33051153c35ee0fc16353e0f649aa

**Comment by MarcoFalke** [comment]
utACk 195054096aa33051153c35ee0fc16353e0f649aa

**Comment by hebasto** [comment]
@practicalswift would you mind re-reviewing this PR?

**Comment by hebasto** [comment]
@laanwj 
> I think this breaks the functionality of using `--setup` together with other options?

After reconsidering your words, I believe you are right.
Fixed. No breaking changes now.
It is possible to run:
```sh
./gitian-build.py --setup --build ${SIGNER} ${VERSION}
```
or
```sh
./gitian-build.py --setup
```

**Comment by hebasto** [comment]
PR description updated.

**Comment by hebasto** [comment]
Rebased.

**Pull request merged**